src/WAC.php

```diff
@@ -130,6 +164,46 @@ private function getUserGrants($resourcePath, $webId) {
 		return $grants;
 	}
 
+	private function getOriginGrants($resourcePath, $origin) {
+		$aclPath = $this->getAclPath($resourcePath);
+		if (!$aclPath) {
+			return array();
+		}
+		$acl = $this->filesystem->read($aclPath);
+
+		$graph = new \EasyRdf_Graph();
```

Comment:
The `new \EasyRdf_Graph()` and `$_SERVER['REQUEST_SCHEME']` should _eventually_ come from outside the class, but fine for now.